src/containers/qtreetbl.c

```diff
@@ -1050,6 +1195,11 @@ static qtreetbl_obj_t *put_obj(qtreetbl_t *tbl, qtreetbl_obj_t *obj,
     if (is_red(obj->left) && is_red(obj->left->left)) {
         obj = rotate_right(obj);
     }
+    
+    // split 4-nodes on the way up.
+    /*if (is_red(obj->left) && is_red(obj->right)) {
```

Comment:
can we remove this if this is not needed?

Reply:
I removed these comments in the last commit.